server.js

```diff
@@ -1,30 +1,74 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
-//   PORT=9000 npm start
+// PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+const listEndpoints = require('express-list-endpoints')
+
 // Start defining your routes here
+
+// Root endpoint
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Route for array of all books with pagination as default
+app.get('/books', (req, res) => {
+  const { page, sort, author, title } = req.query
+  let booksList = booksData
+
+  // sort books by rating using query, returns sorted booksList
+  if (sort === 'rating_dsc') {
+    booksList = booksList.sort((a, b) => (b.average_rating - a.average_rating))
+  } else if (sort === 'rating_asc') {
+    booksList = booksList.sort((a, b) => (a.average_rating - b.average_rating))
+  }
+
+  // find books by author using query, returns filtered booksList
+  if (author) {
+    booksList = booksList.filter((item) => item.authors.toString().toLowerCase().includes(author.toLowerCase()))
+  }
+
+  // find books based on words in title using query, returns filtered booksList
+  if (title) {
+    booksList = booksList.filter((item) => item.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+
+  // slice used for pagination
+  const startIndex = 10 * (+page - 1) || 0
+  const endIndex = startIndex + 10
+  const booksListPaginated = booksList.slice(startIndex, endIndex)
```

Comment:
The `startIndex` could use a comment explaining the math going on there 👍

Reply:
Added some comments 😊